src/test/scala/chiselTests/LFSR16.scala

```diff
@@ -6,43 +6,45 @@ import chisel3._
 import chisel3.testers.BasicTester
 import chisel3.util._
 
-class LFSR16 extends Module {
-  val io = IO(new Bundle {
-    val inc = Input(Bool())
-    val out = Output(UInt(16.W))
-  })
-  val res = RegInit(1.U(16.W))
-  when (io.inc) {
-    val nxt_res = Cat(res(0)^res(2)^res(3)^res(5), res(15,1))
-    res := nxt_res
+/**
+  * This test creates two 4 sided dice.
+  * each cycle it adds them together and adds a count to the bin corresponding to that value
+  * The asserts check that the bins show the correct distribution.
+  */
+//noinspection TypeAnnotation
+//scalastyle:off magic.number
+class LSFRTester extends BasicTester {
+  val bins = Reg(Vec(8, UInt(32.W)))
+
+  // Use tap points on each LFSR so values are more independent
+  val die0 = Cat(Seq.tabulate(2) { i => LFSR16()(i) })
+  val die1 = Cat(Seq.tabulate(2) { i => LFSR16()(i + 2) })
+
+  val (trial, done) = Counter(true.B, 100000)
```

Comment:
How long does this test take to run?

Reply:
This takes no time to run. The run-time is negligible compared to the verilator compile time.
I could shorten if you think it's important but I've tried it at even 10 times more and it still takes
no time.

Reply:
Shortened it by factor of 10